examples/2d/text_input_2d.rs

```diff
@@ -0,0 +1,475 @@
+fn main() {
+    let mut app = App::new();
+
+    app.add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(
+            PostUpdate,
+            (update_text_input_attributes, update_inputs, update_targets).before(TextInputSystems),
+        );
+    app.sub_app_mut(RenderApp)
+        .add_systems(ExtractSchedule, extract_text_input);
+
+    app.run();
+}
+
+fn setup(mut commands: Commands) {
+    commands.spawn(Camera2d);
+
+    let submit_target = commands
+        .spawn((
+            Text2d::new("submit with SHIFT + ENTER"),
+            Anchor::TOP_CENTER,
+            TextBounds {
+                width: Some(500.),
+                height: None,
+            },
+            TextLayout {
+                linebreak: LineBreak::AnyCharacter,
+                justify: Justify::Left,
+            },
+            Transform::from_translation(Vec3::new(0., -25., 0.)),
+        ))
+        .id();
+
+    commands
+        .spawn((
+            TextInputBuffer {
+                cursor_blink_timer: Some(0.),
+                ..Default::default()
+            },
+            TextLayoutInfo::default(),
+            TextEdits::default(),
+            TextInputAttributes::default(),
+            TextInputTarget::default(),
+            Overwrite::default(),
+            TextFont::default(),
+            TextInputSize(Vec2::new(500., 250.)),
+            Transform::from_translation(Vec3::new(0., 150., 0.)),
+            Placeholder::new("type here.."),
+            Visibility::default(),
+            VisibilityClass([TypeId::of::<Sprite>()].into()),
+            Anchor::CENTER,
+        ))
+        .observe(
+            move |event: On<TextInputEvent>, mut query: Query<&mut Text2d>| {
+                if let TextInputEvent::Submission { text } = event.event()
+                    && let Ok(mut target) = query.get_mut(submit_target)
+                {
+                    target.0 = text.clone();
+                }
+            },
+        );
+}
+
+fn update_targets(
+    mut commands: Commands,
+    windows: Query<&Window, With<PrimaryWindow>>,
+    mut query: Query<(
+        Entity,
+        &TextInputSize,
+        &mut TextInputTarget,
+        &Anchor,
+        Option<&mut Aabb>,
+    )>,
+) {
+    let scale_factor = windows
+        .single()
+        .map(|window| window.resolution.scale_factor())
+        .unwrap_or(1.0);
+    for (entity, size, mut target, anchor, aabb) in query.iter_mut() {
+        if target.set_if_neq(TextInputTarget {
+            size: size.0 * scale_factor,
+            scale_factor,
+        }) {
+            let (x1, x2, y1, y2) = anchor.reposition(size.0);
+            let new_aabb = Aabb::from_min_max(Vec3::new(x1, y1, 0.), Vec3::new(x2, y2, 0.));
+
+            if let Some(mut aabb) = aabb {
+                *aabb = new_aabb;
+            } else {
+                commands.entity(entity).try_insert(new_aabb);
+            }
+        }
+    }
+}
+
+fn update_text_input_attributes(
```

Comment:
why is this system needed? I don't think anything is changing the `TextFont` component